shared/src/ledger/eth_bridge/mod.rs

```diff
@@ -44,16 +49,106 @@ where
 
     fn validate_tx(
         &self,
-        _tx_data: &[u8],
-        _keys_changed: &BTreeSet<Key>,
-        _verifiers: &BTreeSet<Address>,
+        tx_data: &[u8],
+        keys_changed: &BTreeSet<Key>,
+        verifiers: &BTreeSet<Address>,
     ) -> Result<bool, Self::Error> {
-        tracing::debug!(
-            tx_data_len = _tx_data.len(),
-            keys_changed_len = _keys_changed.len(),
-            verifiers_len = _verifiers.len(),
+        tracing::info!(
+            tx_data_len = tx_data.len(),
+            keys_changed_len = keys_changed.len(),
+            verifiers_len = verifiers.len(),
             "Validity predicate triggered",
         );
+        let signed: Signed<Vec<u8>> =
+            match Signed::<Vec<u8>>::try_from_slice(tx_data) {
```

Comment:
If submitting a protocol transaction, `tx_data` ends up being the `tx.data` field of the innermost `Tx` (i.e. the one wrapped by `EthereumBridgeUpdate(Tx)`), which should be a `Signed { data, sig }` if the protocol transaction is constructed correctly. 

In the case of submitting a regular transaction, it seems to end up being a `SignedTxData`, which I think must be the `tx.data` field of the outermost `Tx` that actually gets submitted to Tendermint RPC. 

I haven't looked closely yet at why there is this difference, it could be just to do with the way this branch is coded currently, but the VP will reject as it can't deserialize that to a `Signed`, if it gets that far.